[TEST] conda smithy update-cb3

shadowwalkersb opened the pull request:
Testing `conda smithy update-cb3`.

```
$ conda smithy update-cb3

List of changes done to the recipe:
Renamed build with host
Moving cmake from host to build
Moving `python # [win]` which was used for vc matrix
Moving msinttypes from host to build
Removing pinnings for openssl to use values from conda_build_config.yaml
Removing pinnings for libiconv to use values from conda_build_config.yaml
Removing pinnings for curl to use values from conda_build_config.yaml
Removing pinnings for libssh2 to use values from conda_build_config.yaml
Moving pkg-config from host to build
Removing vc features
Removing features section as it is empty
Adding C compiler
```

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict.
Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the @ notation in a comment) if you believe this is a bug.

shadowwalkersb committed:
revert to cb2

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Jinja2 variable definitions are suggested to take a ``{%<one space>set<one space><variable name><one space>=<one space><expression><one space>%}`` form. See lines [4, 144, 145, 146]

shadowwalkersb committed:
update to cb3

$ conda smithy update-cb3
List of changes done to the recipe:
Renamed build with host
Moving cmake from host to build
Moving `python # [win]` which was used for vc matrix
Moving msinttypes from host to build
Removing pinnings for openssl to use values from conda_build_config.yaml
Removing pinnings for libiconv to use values from conda_build_config.yaml
Removing pinnings for curl to use values from conda_build_config.yaml
Removing pinnings for libssh2 to use values from conda_build_config.yaml
Moving pkg-config from host to build
Removing vc features
Removing features section as it is empty
Adding C compiler

shadowwalkersb committed:
MNT: Re-rendered with conda-smithy 3.1.5 and pinning 2018.05.22

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

shadowwalkersb committed:
update to cb3

$ conda smithy update-cb3
List of changes done to the recipe:
Renamed build with host
Moving cmake from host to build
Moving pkg-config from host to build
Adding C compiler

shadowwalkersb commented:
@isuruf ,`msinttypes` moved under `host`, but `{{ compiler('c') }}` has been added and is duplicated under both `build` and `host`. Also, the second `host` is a duplicate, too.

isuruf commented:
The script doesn't work if the recipe still uses conda-build=3 features

shadowwalkersb commented:
Is there a condition when C++ compilers will be added or is it always going to be only C compilers?

shadowwalkersb committed:
revert to cb2

shadowwalkersb committed:
update to cb3

$ conda smithy update-cb3
List of changes done to the recipe:
Renamed build with host
Moving cmake from host to build
Moving `python # [win]` which was used for vc matrix
Removing pinnings for openssl to use values from conda_build_config.yaml
Removing pinnings for libiconv to use values from conda_build_config.yaml
Removing pinnings for curl to use values from conda_build_config.yaml
Removing pinnings for libssh2 to use values from conda_build_config.yaml
Moving pkg-config from host to build
Removing vc features
Removing features section as it is empty
Adding C compiler

isuruf commented:
If there's a file with extension `.cpp` or `.cxx` in the source then it will add a C++ compiler.

shadowwalkersb commented:
> If there's a file with extension .cpp or .cxx in the source then it will add a C++ compiler.

Cool.

shadowwalkersb commented on recipe/meta.yaml:
```diff
@@ -38,6 +37,8 @@ requirements:
 
 test:
+  requires:
+    - python {{ environ['PY_VER'] + '*' }}  # [win]
```
Is this line still needed?

shadowwalkersb replied:
This what gets installed in `vc9` test environment (https://ci.appveyor.com/project/conda-forge/libgit2-feedstock/build/1.0.81/job/pytyi6qnjtk0t6ex#L1889) when this is present, mixture of `vc9` and `vc14`.

```
## Package Plan ##
  environment location: C:\bld\libgit2_1528674472804\_test_env
The following NEW packages will be INSTALLED:
    curl:           7.60.0-vc9_0      conda-forge [vc9 ]
    krb5:           1.14.6-vc9_0      conda-forge [vc9 ]
    libgit2:        0.27.1-h3e54e04_0 local      
    libssh2:        1.8.0-vc14_1      conda-forge [vc14]
    python:         3.6.5-1           conda-forge
    vc:             9-0               conda-forge
    vs2008_runtime: 9.0.30729.6161-0  conda-forge
    vs2015_runtime: 14.0.25420-0      conda-forge
    zlib:           1.2.11-vc14_0     conda-forge [vc14]
```

shadowwalkersb committed:
remove test/requires: python {{ environ['PY_VER'] + '*' }}  # [win]

shadowwalkersb commented:
And, this is what we get without `python {{ environ['PY_VER'] + '*' }} # [win]`, https://ci.appveyor.com/project/conda-forge/libgit2-feedstock/build/1.0.82/job/m1ya4wqxf6tqbxyw#L1889.

```
## Package Plan ##
  environment location: C:\bld\libgit2_1528698806029\_test_env
The following NEW packages will be INSTALLED:
    ca-certificates: 2018.4.16-0       conda-forge
    curl:            7.60.0-vc9_0      conda-forge [vc9]
    krb5:            1.14.6-vc9_0      conda-forge [vc9]
    libgit2:         0.27.1-h3e54e04_0 local      
    libssh2:         1.8.0-vc9_2       conda-forge [vc9]
    openssl:         1.0.2o-vc9_0      conda-forge [vc9]
    vc:              9-0               conda-forge
    vs2008_runtime:  9.0.30729.6161-0  conda-forge
    zlib:            1.2.11-vc9_0      conda-forge [vc9]
```

So, I don't know if this is a conda-build issue or should those lines be removed by conda-smithy as part of the upgrade to cb3?

jakirkham commented:
Those should probably be removed by `conda-smithy` as part of the upgrade. Problem is we have a lot of variety in terms of that line.

shadowwalkersb commented:
> Problem is we have a lot of variety in terms of that line.

I can imagine... :-(
